server/src/Level.ts

```diff
@@ -1,36 +1,110 @@
+import { ToServerEvents, ToClientEvents } from "common/messages/sceneGameArena";
+import { broadcastFallRate } from "../index";
+import { Socket } from "socket.io";
+
+type SocketLevel = Socket<ToServerEvents, ToClientEvents>;
+
 export class Level {
     private _currentLevel: number;
+    private currentFallRate: number;
 
     constructor() {
         this._currentLevel = 1;
+        this.currentFallRate = 1000;
     }
 
     get currentLevel(): number {
         return this._currentLevel;
     }
 
+    /**
+     * Setup listeners for the current client socket.
+     * @param socket The client's socket.
+     */
+    public initSocketListeners(socket: SocketLevel) {
+        socket.on("requestFallRate", () => {
+            socket.emit("updateFallRate", this.currentFallRate);
+        });
+    }
+
+    /**
+     * Check if its possible to update the game's level based off the given score. Max level is 15.
+     * @param score The total score the players have accumulated.
+     */
     public checkUpdateLevel(score: number) {
-        if (score >= this._currentLevel * 20) {
+        if (
+            score - (this._currentLevel - 1) * 20 >= this._currentLevel * 20 &&
+            this._currentLevel < 15
+        ) {
```

Comment:
This still doesn't comply with the SRS due to score rollover (as alluded to in my previous comment)

From SRS 4.8.3:
> Additional points scored above a given level’s goal do not contribute to the goal of the next level.

Reply:
I've renamed the _totalScore in the Scoreboard.ts file to now be _accumualtedScore.
It represents the accumulated score for the current level and is reset whenever the level increases.